tests/test_analysis.py

```diff
@@ -0,0 +1,95 @@
+import covasim as cv
+import numpy as np
+
+
+# Suite of tests to test basic functionality of the analysis.py file
+# Runtime: 5.004 seconds
+def test_analysis_snapshot():
+    sim = cv.Sim(analyzers=cv.snapshot('2020-04-04', '2020-04-14'))
+    sim.run()
+    snapshot = sim['analyzers'][0]
+    people1 = snapshot.snapshots[0]            # Option 1
+    people2 = snapshot.snapshots['2020-04-04'] # Option 2
+    people3 = snapshot.get('2020-04-14')       # Option 3
+    people4 = snapshot.get(34)                 # Option 4
+    people5 = snapshot.get()                   # Option 5
+    # people3 = []  # uncomment to verify error
+    peoples = [people1, people2, people3, people4, people5]
+    for i, people in enumerate(peoples):
+        optionNum = i+1
+        assert len(people) > 0, f"Option {optionNum} should have more than 0 members"
+
+
+
+def test_analysis_hist():
+    # raising multiple histograms to check windows functionality
+    day_list = ["2020-03-30", "2020-03-31", "2020-04-01"]
+    age_analyzer = cv.age_histogram(days=day_list)
+    sim = cv.Sim(analyzers=age_analyzer)
+    sim.run()
+    assert age_analyzer.window_hists == None

```

Comment:
I wouldn't check this implementation detail...this could change to, say, an empty list in future

Reply:
How about '''assert not age_analyzer.window_hists? '''

Reply:
I think I'd prefer to avoid this check since (a) in future it might change to populate this when the sim is run in which case the assert would fail, (b) this is a somewhat experimental feature (it's not mentioned in the documentation, for example) and might be renamed/changed in future.

Reply:
So long as we keep some kind of assertion in the test that is predictive (in this case the final assertion that the number of histograms / plots produced is equal to the number of days requested) then we're probably okay. After chewing on this one a bit, I suppose the critical bit here is that you get the windows and plots that you expect to get. At some point I'm assuming it would be appropriate to add other verifications that (for example) "the number of symptomatic people in the 18-25 age bucket on day x-y-z" is the same in the simulation report channels as it is in the histogram, but this test doesn't need to be as precise.